### PR TITLE
Chore/promise tests speed optimization

### PR DESCRIPTION
**Pull request by gentique**
Before optimization test duration:
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 24.470 (24.523) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 20.552 (20.604) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 24.469 (24.520) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 23.518 (23.569) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 22.065 (22.119) seconds
On average: 22,6 seconds

After optimization test duration:
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 1.933 (1.978) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 1.884 (1.925) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 1.893 (1.944) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 1.888 (1.932) seconds
	 - 	 Executed 127 tests, with 0 failures (0 unexpected) in 1.920 (1.966) seconds
On average: 1,9 seconds

A **91,59%** improvement in test time.

### CODE REVIEW

File: Tests/Tests/PromiseKit/PromiseTests.swift [hunk 1]
```diff
@@ -102,14 +102,14 @@ class PromiseTests: XCTestCase {
   func testChainResolvedOnMainThread() {
     let ex1 = expectation(description: "")
     let ex2 = expectation(description: "")
-    async(0)
-      .flatMap(on: .main) { async($0, on: .main) }
+    async(0, delay: 0.01)
```
[the remark this answers is not in the document]
Comment (gentique) [reply Tests/Tests/PromiseKit/PromiseTests.swift:105]: Hmm I am not sure if I'll introduce undesired behavior in projects who rely on that 0.5 delay. It seemed safer to do it this way.

Though I am open to do that, I could drop the previous commits and fix it with a new one where I'll change the default delay.

[the remark this answers is not in the document]
Comment (tonikocjan) [reply Tests/Tests/PromiseKit/PromiseTests.swift]: Great, good work. 👍

**Comment by borut-t** [comment]
I remember we had some issues with shorter delays. Are you sure this will hold?

**Comment by tonikocjan** [comment]
@borut-t we had issues with `waitForExpectation. This should be okay.

**Comment by gentique** [comment]
@borut-t Also if it were to fail due to a shorter delay it would indicate that something in the implementation is wrong.

**Review by borut-t — APPROVED** [review]
👌

**Pull request merged**